### ISSUE TITLE
`single-use-lifetimes` false positive

### ISSUE THREAD

**Issue by AntoniosBarotsis** [rust-lang/rust-clippy#10054]
### Summary

I got the warning mentioned below which did not make sense to me. I'm rather new to Rust so I asked if this was indeed a false positive in the Rust Discord and people seemed to agree so here I am.

### Lint Name

single-use-lifetimes

### Reproducer

I tried this code:

```rust
#[derive(Debug, PartialEq)]
enum Command<'a> {
  Cd(&'a str, Vec<&'a str>),
  Ls,
}
```

I saw this happen:

```
warning: lifetime parameter `'a` only used once
  --> src\bin\07.rs:41:14
   |
40 | #[derive(Debug, PartialEq)]
   |                 --------- ...is used only here
41 | enum Command<'a> {
   |              ^^ this lifetime...
   |
   = note: requested on the command line with `-W single-use-lifetimes`
help: elide the single-use lifetime
   |
40 ~ #[derive(Debug, '_)]
41 ~ enum Command {
   |

warning: `advent_of_code` (bin "07") generated 1 warning
```

I expected to see this happen: No warning

### Version

```text
rustc 1.65.0 (897e37553 2022-11-02)
binary: rustc
commit-hash: 897e37553bba8b42751c67658967889d11ecd120
commit-date: 2022-11-02
host: x86_64-pc-windows-msvc
release: 1.65.0
LLVM version: 15.0.0
```

**Comment by Jarcho** [comment]
`single_use_lifetimes` is a rustc lint, not a clippy lint. This is also fixed in 1.66 which is currently in beta.
